src/babylonjs/renderer/loaders/GLTFLoader.js

```diff
@@ -13,6 +16,57 @@ export class GLTFLoader extends LoaderBase {
 
 	}
 
+	/**
+	 * Detect if buffer contains GLB binary data by checking magic bytes
+	 * @param {ArrayBuffer|Uint8Array} buffer - The file buffer
+	 * @returns {boolean} True if GLB format
+	 */
+	isGLB( buffer ) {
+
+		// Handle both ArrayBuffer and typed arrays (e.g., Uint8Array)
+		const arrayBuffer = buffer instanceof ArrayBuffer ? buffer : buffer.buffer;
+		const byteOffset = buffer instanceof ArrayBuffer ? 0 : buffer.byteOffset;
+		const byteLength = buffer.byteLength;
+
+		if ( byteLength < 4 ) {
+
+			return false;
+
+		}
+
+		const view = new DataView( arrayBuffer, byteOffset, byteLength );
+		const magic = view.getUint32( 0, true ); // little-endian
+		return magic === GLB_MAGIC;
+
+	}
```

Comment:
Just to be sure: there's no way for Babylon to determine whether a file is a binary or ascii glTF automatically?